pkg/controller/computenodeopenstack/computenodeopenstack_controller.go

```diff
@@ -395,45 +483,114 @@ func updateNodesStatus(c client.Client, instance *computenodev1alpha1.ComputeNod
 				err = deleteBlockerPodFinalizer(c, node.Name, instance)
 			}
 		case "Ready":
+			log.Info("Previous Status Ready: " + node.Name)
 			newNode := computenodev1alpha1.Node{Name: node.Name, Status: nodeStatus}
 			nodesStatus = append(nodesStatus, newNode)
 
 			if nodeStatus == "Ready" {
 				readyWorkers += 1
 			} else if nodeStatus == "SchedulingDisabled" {
-				// trigger drain (new pod)
-				err = triggerNodeDrain(c, node.Name, instance)
+				// trigger node drain pre tasks (new pod)
+				err = triggerNodeDrain(c, node.Name, instance, true, openstackClientAdmin, openstackClient)
 				if err != nil {
 					return err
 				}
 			}
 		case "SchedulingDisabled":
-			// if drain has finished (pod completed) remove blocker-pod finalizer
-			drained, err := isNodeDrained(c, node.Name, instance)
-			if err != nil {
+			log.Info(fmt.Sprintf("Previous Status SchedulingDisabled: %s", node.Name))
+			// if pre draining has finished (job completed) remove blocker-pod finalizer
+			drainedPre, err := isNodeDrained(c, node.Name, node.Name+"-drain-job-pre", instance)
+			if err != nil && errors.IsNotFound(err) {
+				log.Error(err, fmt.Sprintf("Pre NodeDrain job errors.IsNotFound: %s", node.Name+"-drain-job-pre"))
+				ResyncPeriod = 30 * time.Second
+				return nil
+			} else if err != nil {
 				return err
 			}
-			if drained {
-				err = deleteBlockerPodFinalizer(c, node.Name, instance)
+
+			log.Info(fmt.Sprintf("Pre draining job: %s", drainedPre))
+			switch drainedPre {
+			case "succeeded":
+				// taint node that daemonset pods get stopped
+				nodes, err := kclient.CoreV1().Nodes().List(metav1.ListOptions{})
+
+				for _, updateNode := range nodes.Items {
+					log.Info("NodeDrainPre updateNode: " + updateNode.Name)
+					if updateNode.Name == node.Name {
+						log.Info("NodeDrainPre our node: " + updateNode.Name)
+
+						err = addToBeRemovedTaint(kclient, updateNode)
+						if err != nil {
+							return err
+						}
+					}
+				}
+
+				// wait until all pods in openstack namespace got shutdown
```

Comment:
can you add a comment about why this is needed? (I actually forgot about it already...)

Reply:
on a second thought... why we need this? What was the problem with the next:
- Predrain
- DeleteBlockerPodFinalizer (so that we don't care about openstack pod removals)
- PostDrain + (NodesToDelete update)

Reply:
when we delete the blocker pod, the node gets removed. The follow up POST jobs, which run longer then the node get deleted won't get recognized any longer. I mean if the node is fully gone, especially when we scale down to 0 the updatesNodesStatus won't get rerun when jobs finish as the updateNodesStatus loops through the NodeList. Therefore to have a clean workflow, we need 1) taint the node to have pods stopped 2) run post drain job 3) when done remove blocker pod